unit_tests/tests/test_call.rs

```diff
@@ -0,0 +1,299 @@
+ * 
+ * purpose: function request `balanceOf` to StarkGate ETH bridge contract
+ * success case: must return non-zero balance
+ */
+#[rstest]
+#[tokio::test]
+async fn work_correct_call_with_args(clients: HashMap<String, JsonRpcClient<HttpTransport>>) {
+    let deoxys = &clients[DEOXYS];
+    let pathfinder = &clients[PATHFINDER];
+
+    let response_deoxys = deoxys.call(
+        FunctionCall { 
+            contract_address: FieldElement::from_hex_be(STARKGATE_ETH_BRIDGE_ADDR).unwrap(), 
+            entry_point_selector: get_selector_from_name("balanceOf").unwrap(),
+            calldata: vec![
+                FieldElement::from_hex_be(CONTRACT_ADDR).unwrap()
+            ]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.expect("Error waiting for response from Deoxys node");
+
+    let response_pathfinder = pathfinder.call(
+        FunctionCall { 
+            contract_address: FieldElement::from_hex_be(STARKGATE_ETH_BRIDGE_ADDR).unwrap(), 
+            entry_point_selector: get_selector_from_name("balanceOf").unwrap(),
+            calldata: vec![
+                FieldElement::from_hex_be(CONTRACT_ADDR).unwrap()
+            ]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.expect("Error waiting for response from Pathfinder node");
+
+    let balance_deoxys = u128::try_from(response_deoxys[0]).unwrap();
+
+    assert!(balance_deoxys > 0);
+    assert_eq!(response_deoxys, response_pathfinder);
+}
+
+/**
+ * Unit test for `starknet_call`
+ * 
+ * purpose: function request `sort_tokens` to JediSwap exchange, with multiple arguments.
+ * success case: must return array of 2 non-zero values.
+ */
+#[rstest]
+#[tokio::test]
+async fn work_with_multiple_args(clients: HashMap<String, JsonRpcClient<HttpTransport>>) {
+    let deoxys = &clients[DEOXYS];
+    let pathfinder = &clients[PATHFINDER];
+
+    let response_deoxys = deoxys.call(
+        FunctionCall {
+            contract_address: FieldElement::from_hex_be(JEDI_SWAP_ADDR).unwrap(),
+            entry_point_selector: get_selector_from_name("sort_tokens").unwrap(),
+            calldata: vec![
+                FieldElement::from_hex_be(STARKGATE_ETHER).unwrap(),
+                FieldElement::from_hex_be(STARKGATE_USDC).unwrap()
+            ]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.expect("Error waiting for response from Deoxys node");
+
+    let response_pathfinder = pathfinder.call(
+        FunctionCall {
+            contract_address: FieldElement::from_hex_be(JEDI_SWAP_ADDR).unwrap(),
+            entry_point_selector: get_selector_from_name("sort_tokens").unwrap(),
+            calldata: vec![
+                FieldElement::from_hex_be(STARKGATE_ETHER).unwrap(),
+                FieldElement::from_hex_be(STARKGATE_USDC).unwrap()
+            ]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.expect("Error waiting for response from Deoxys node");
+
+    assert!(response_deoxys.len() == 2);
```

Comment:
Same could you detail here

Reply:
Here again, tests are being made to make sure Deoxys complies to the signature of the [function](https://starkscan.co/token/0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7)

1. Function returns two arguments.
2. Function returns non-zero felt values

Like in the previous case, once it has been validated that Deoxis' response matches the required format, it is then compared to the answer by the Pathfinder node to verify exactitude.